### DISCUSSION TITLE
月曜日に投稿した記事ほどランキングメールに載りやすい問題について

### DISCUSSION THREAD

**Discussion by JunichiIto** [increments/qiita-discussions#280] [要望・アイデア・不具合]
以前から密かに気になってた問題なのですが、毎週水曜日に送られてくるランキングメールは月曜開始、日曜終了のLGTM数をカウントしています。

<img width="402" alt="Screen Shot 2022-08-22 at 16 19 30" src="https://user-images.githubusercontent.com/1148320/185862303-d048c3d2-5e7f-4f2b-a40e-512fe6dce5e0.png">

そのため、月曜日に投稿した記事が一番ランキングに載りやすく、日曜日に投稿した記事が一番ランキングに載りにくい問題があります。

例（集計期間が8/8-8/14、ランクインの最低ラインが70LGTMだった場合）

- 記事A 8/8(月)に記事公開、8/14までに合計70LGTMを獲得 → 70LGTMなのでランクインする
- 記事B 8/14(日)に記事公開、8/14に50LGTM、8/15に40LGTMを獲得 → 2日間で90LGTMだが、集計期間内で見ると50LGTMなのでランキング外

ときどき日曜日に記事を書いたときに「これ、今日公開する？それとも明日まで待った方がいい？？」と妙な自問自答が生まれてしまうので、どの曜日に投稿しても平等に評価されるランキングになっていると嬉しいです。

具体的にどうするの？って言われるとノーアイデアなのですが・・・。すいません。

**Comment by tanaka-asami415** [comment]
@JunichiIto 
とんでもないです！ご意見ありがとうございます🙏

集計期間の変更などの改善方法はあるかもしれないのですが、投稿してから1週間以上の記事の扱いなど、
まだ、ベストだという方法が思い浮かばず…もう少し検討する必要がありそうです。

すぐの改善は難しいと思いますが、引き続き検討したいと思います🙏
検討する必要があることを気づかせていただき、ありがとうございます🙇

**Reply by JunichiIto** [reply]
返信ありがとうございます。
僕も「こういう集計方法がいいと思いますよ」という提案できたらいいんですが。。
とはいえ、「いい記事なんだけど、日曜に投稿したせいでランクインできずに埋もれていった」という記事があるともったいないので、この件が解決されると投稿者としては大変ありがたいです。
ご検討よろしくお願いします🙏

**Comment by munierujp** [comment]
素人アイデアですが、集計期間を固定にするのではなく、各記事が公開されてから1週間の評価をもとにランキングをつくるなどすれば公平になりそうな気がします。

**Reply by JunichiIto** [reply]
僕も最初それを考えたんですが、新規に投稿された記事はそれでいいものの、公開後2週目以降のLGTMはどうカウントするの？みたいなことを考え出すと、結構ややこしくなりそうな気がします。
（日曜日に投稿された記事は2週目以降も引き続き日曜始まりで集計するの？みたいな）

**Reply by ymdarudy** [reply]
悩ましいですね...🤔

下記、思いつきですが一案として。


- 投稿初週のものだけのランキングを別で設ける。
- その上で、日曜日締めは変えないが、「締め時間と投稿時間の差分」から算出した「時間あたりのLGTM数」で初週ランキングを決める。

例)
記事A:
締め時間の30時間前に投稿、LGTM60
→時間あたり2LGTM

記事B:
締め時間の100時間前に投稿、LGTM150
→時間あたり1.5LGTM

Aの記事の方が上位あつかい。
